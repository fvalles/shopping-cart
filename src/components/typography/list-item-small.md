Basic list item small text component with default color="summaryText", font-weight="normal" and margin=0:

```jsx
<ListItemSmall>Lorem ipsum</ListItemSmall>
```

List item small text component with color="primary", font-weight="bold" and margin="0px 0px 0px 2px":

```jsx
<ListItemSmall color="primary" fontWeight="bold" margin="0px 0px 0px 2px">
  Lorem ipsum
</ListItemSmall>
```
