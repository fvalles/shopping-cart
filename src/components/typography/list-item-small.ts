import styled, { css } from 'styled-components';
import { TypographyProps } from './common';

export const ListItemSmall = styled.span<TypographyProps>`
  color: ${({ color = 'summaryText', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.light}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 14px;
  ${({ fontWeight }) =>
    fontWeight &&
    css`
      font-weight: ${fontWeight};
    `}
  line-height: 17px;
  ${({ margin }) =>
    margin &&
    css`
      margin: ${margin};
    `}
`;
