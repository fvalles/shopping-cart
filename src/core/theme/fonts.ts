export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
