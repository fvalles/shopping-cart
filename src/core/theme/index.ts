export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
