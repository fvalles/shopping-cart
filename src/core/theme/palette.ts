export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
