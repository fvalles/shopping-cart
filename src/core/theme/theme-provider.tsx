import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
