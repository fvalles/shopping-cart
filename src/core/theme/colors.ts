import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
