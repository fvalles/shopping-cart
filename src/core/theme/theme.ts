import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
