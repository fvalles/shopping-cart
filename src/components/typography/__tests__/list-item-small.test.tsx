import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ListItemSmall } from '../list-item-small';

describe('list item small component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <ListItemSmall>Lorem ipsum</ListItemSmall>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#212240');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Light,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '14px');
    expect(result).toHaveStyleRule('font-weight', 'normal');
    expect(result).toHaveStyleRule('line-height', '17px');
    expect(result).toHaveStyleRule('margin', '0');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom white color, 16px margin and font-weight bold', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <ListItemSmall color="white" fontWeight="bold" margin="16px">
            Lorem ipsum
          </ListItemSmall>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#FFFFFF');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Light,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '14px');
    expect(result).toHaveStyleRule('font-weight', 'bold');
    expect(result).toHaveStyleRule('line-height', '17px');
    expect(result).toHaveStyleRule('margin', '16px');
    expect(result).toMatchSnapshot();
  });
});
