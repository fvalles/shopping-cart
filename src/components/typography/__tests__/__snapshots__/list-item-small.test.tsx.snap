// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list item small component should render with custom white color, 16px margin and font-weight bold 1`] = `
.c0 {
  color: #FFFFFF;
  font-family: Avenir Light,'Avenir Next','Segoe UI', sans-serif;
  font-weight: bold;
  margin: 16px;
  font-size: 14px;
  line-height: 17px;
}

<span
  className="c0"
  color="white"
  fontWeight="bold"
>
  Lorem ipsum
</span>
`;

exports[`list item small component should render with default css properties 1`] = `
.c0 {
  color: #212240;
  font-family: Avenir Light,'Avenir Next','Segoe UI', sans-serif;
  font-weight: normal;
  margin: 0;
  font-size: 14px;
  line-height: 17px;
}

<span
  className="c0"
>
  Lorem ipsum
</span>
`;
